adr/2025_06_13-document-transforms.md

```diff
@@ -0,0 +1,65 @@
+# Document Transforms
+
+A core part of the BlockNote API is the ability to make changes to the document, using BlockNote's core design of blocks, this is much easier to do programmatically than with other editors.
+
+We've done pretty well with our existing API, but there are a few things that could be improved.
+
+Referencing content within the document is either awkward or non-existent. Right now we essentially really only have an API for referencing blocks by their id with no further level of granularity.
```

Comment:
Overall, I think the concept of a `Location` is useful to have - but if we start implementing this I think we should have more sight on practical use-cases. i.e. which use-cases /  APIs do we want to "unlock" with this?

Reply:
insertContentAt(at: Location, content: Block | PartialBlock | string)
getSelection(): {range: Range}

A number of the low-level Tiptap API uses for things like deleting some chars

This will also be very useful for server-side editing ops like rewriting a paragraph